rust/projects/file-io/tests/tests.rs

```diff
@@ -113,12 +142,29 @@ fn binary_path() -> PathBuf {
     path
 }
 
+fn temp_dir(name: &str) -> PathBuf {
+    let mut temp_dir = env::temp_dir();
+    temp_dir.push(name);
+    create_dir_all(&temp_dir).expect("unable to create working directory");
+    temp_dir
+}
```

Comment:
From the looks of this the temp dirs are not deleted up after the test. Can you instead use the tempfile crate's tempdir, which has destructors that automatically clean up.

Reply:
I was avoiding introducing dependencies for tests. If we do so, we'll either change to use workspaces or just add dev-dependencies to Cargo.toml (maybe mention that in project 1).

Reply:
This way, students copy the content of Cargo.toml at first, and copy the tests of each project, then everything will go fine.